src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestHandler.cs

```diff
@@ -115,26 +145,59 @@ public void ProcessRequests()
                         var areTestCaseLevelEventsRequired = this.dataCollectionManager.SessionStarted();
 
                         // Open a socket communication port for test level events.
-                        int testCaseEventsPort = -1;
+                        var testCaseEventsPort = 0;
                         if (areTestCaseLevelEventsRequired)
                         {
                             testCaseEventsPort = this.dataCollectionTestCaseEventHandler.InitializeCommunication();
 
-                            this.testCaseEventMonitorTask = Task.Factory.StartNew(() =>
-                             {
-                                 // todo : decide the time out for this wait as it has to wait till test execution process get created.
-                                 this.dataCollectionTestCaseEventHandler.WaitForRequestHandlerConnection(0);
-                                 this.dataCollectionTestCaseEventHandler.ProcessRequests();
-                             }, this.cancellationTokenSource.Token);
+                            this.testCaseEventMonitorTask = Task.Factory.StartNew(
+                                () =>
+                                    {
+                                        try
+                                        {
+                                            if (this.dataCollectionTestCaseEventHandler.WaitForRequestHandlerConnection(dataCollectionConnTimeout))
```

Comment:
Do we support `VSTEST_COLLECTOR_DEBUG` for debugging data collectors? If yes, we need to add a note to user here that data collector process is waiting for debugger attach.